tests/resources/Harbor-Pages/Replication.robot

```diff
@@ -20,25 +20,81 @@ Resource  ../../resources/Util.robot
+    #input necessary info
+    Input Text  xpath=${destination_name_xpath}  ${name}
+    Input Text  xpath=${destination_url_xpath}  ${url}
+    Input Text  xpath=${destination_username_xpath}  ${username}
+    Input Text  xpath=${destination_password_xpath}  ${pass}
+    #cancel verify cert since we use a selfsigned cert
+    Click Element  ${destination_insecure_xpath}
+    #test connection,here should be success
+    #Click Element  //button[contains(.,'Test')]
+    #Wait Until Page Contains  successfully
+    Click Element  //button[contains(.,'OK')]
+    #verify
+    Wait Until Page Contains  ${name}
 
-    Click element  ${new_name_xpath}
-    Sleep  2
-    	
-    Input Text  xpath=${policy_name_xpath}  ${policy_name}
-    Input Text  xpath=${policy_description_xpath}   ${policy_description}
+Create A Rule With Existing Endpoint
+    [Arguments]  ${name}  ${project_name}  ${endpoint}  ${mode}
+    #click new
+    Click Element  //button[contains(.,'New')]
+    #input name
+    Input Text  //input[@id='ruleName']  ${name}
+    #input descripiton,here skip, leave it blank
+    #source projects, input
+    Input Text  //input[@value='name']  ${project_name}
+    #set filter
+    Click Element  //hbr-create-edit-rule/clr-modal/div/div[1]/div/div[1]/div/div[2]/form/section/div[4]/clr-icon
+    Input Text  //hbr-create-edit-rule//section/div[4]/div/div[1]/div/label/input  *
+    Click Element  //hbr-create-edit-rule/clr-modal/div/div[1]/div/div[1]/div/div[2]/form/section/div[4]/clr-icon
+    Input Text  //hbr-create-edit-rule//section/div[4]/div/div[2]/div/label/input  *
+    #select endpoint
+    Click Element  //select[@id='ruleTarget']
+    Wait Until Element Is Visible  //select[@id='ruleTarget']//option[contains(.,'${endpoint}')]
+    #Select From List By Label  //select[@id='ruleTarget']  ${endpoint}
+    Click Element  //select[@id='ruleTarget']//option[contains(.,'${endpoint}')]
+    #set trigger  
+    Click Element  //select[@id="ruleTrigger"]
+    Wait Until Element Is Visible  //select[@id="ruleTrigger"]//option[contains(.,'${mode}')]
+    Click Element  //select[@id="ruleTrigger"]//option[contains(.,'${mode}')]
+    #click save
+    Click Element  //button[contains(.,'SAVE')]
 
-    #Click element  xpath=${policy_enable_checkbox}
-    #enable attribute is droped in new ui
+Test Endpoint Fail
+    Click Element  //button[contains(.,'Test')]
+    Wait Until Page Contains  Failed
 
-    Click element  xpath=${policy_endpoint_checkbox}
+Cancel Verify Endpoint Cert 
+    Click Element  ${destination_insecure_xpath}
+    Sleep  1
 
-    Click element  xpath=//*[@id="ruleBtnOk"]
-    Sleep  5
-    Capture Page Screenshot  rule_${policy_name}.png
-    Wait Until Page Contains  ${policy_name}
+Test Endpoint Success
+    Click Element  //button[contains(.,'Test')]
+    Wait Until Page Contains  successfully
 
-    Wait Until Page Contains  ${policy_description}
-    Wait Until Page Contains  ${destination_name}
+Create Endpoint Without Saving
```

Comment:
can you bind these two methods(Create Endpoint Without Saving, Create A New Endpoint ) together with one more parameter?